server/selection.go

```diff
@@ -59,6 +61,28 @@ type stakeReader interface {
 	Stakes(addrs []ethcommon.Address) (map[ethcommon.Address]*big.Int, error)
 }
 
+type storeStakeReader struct {
+	store common.OrchestratorStore
+}
+
+func (r *storeStakeReader) Stakes(addrs []ethcommon.Address) (map[ethcommon.Address]*big.Int, error) {
+	orchs, err := r.store.SelectOrchs(&common.DBOrchFilter{Addresses: addrs})
```

Comment:
If I'm not mistaken we only care about active O's here so we can add a currentRound filter to `DBOrchFilter`. 

We're only caching stake for active O's anyway

Reply:
I think the caller of `Stakes()` should receive the stake values for all the addresses that it provides and that it should be up to the caller to decide which addresses that it cares about (i.e. only active addresses). Currently, the caller of `Stakes()` is `MinLSSelector` which will only be populated with active Os when the node is running in on-chain mode. So, in this case, only active addresses will be provided to this method.